### ISSUE TITLE
charm release should have k8s version

### ISSUE THREAD

**Issue by rimusz** [juju-solutions/bundle-canonical-kubernetes#159]
charm release should have k8s version in it's readme.
easier to track and pick the required version to install

**Comment by mbruzek** [comment]
Thanks for the feedback @rimusz. The version of Kubernetes can be found on the `juju status` screen of a running cluster:
```
App                    Version  Status  Scale  Charm                  Store       Rev  OS      Notes
easyrsa                3.0.1    active      1  easyrsa                jujucharms    5  ubuntu     
etcd                   2.2.5    active      3  etcd                   jujucharms   21  ubuntu     
flannel                0.6.1    active      4  flannel                jujucharms    7  ubuntu     
kubeapi-load-balancer  1.10.0   active      1  kubeapi-load-balancer  jujucharms    5  ubuntu  exposed
kubernetes-e2e         1.5.1    active      1  kubernetes-e2e         jujucharms    7  ubuntu     
kubernetes-master      1.5.1    active      1  kubernetes-master      jujucharms   10  ubuntu     
kubernetes-worker      1.5.1    active      3  kubernetes-worker      jujucharms   12  ubuntu  exposed
...
```
The Kubernetes project is fast moving and we decided to decouple the version of the Kubernetes binaries from the charm versions so we can release them independently. You can change the version of Kubernetes by uploading a new resource.  The Kubernetes charms should be able to support an entire minor release of Kubernetes (unless upstream breaks compatibility by adding a new required flag or removing support for a flag - which has happened in the past). So I am not sure how we would keep the current version in sync with the resource as they are versioned independently.

**Comment by marcoceppi** [comment]
We do also track this on the bundle readme, but the charm readmes could be updated to include this as well: https://github.com/juju-solutions/bundle-canonical-kubernetes/blob/master/README.md

**Comment by chuckbutler** [comment]
@marcoceppi @mbruzek @rimusz there's one thing that begs me to say "please dont do this". We attach resources to the charm and can do so retroactively.

The way the charm store works is resources (in this case, the tarballs of kubernetes assets)  are uploaded and get their own resource revision.

Subsequently, we can then attach that resource to any charm revision present in the store, both at time of publish, and retroactively.

So say we have released 1.5.0 with the insecure-by-default bug, and it were attached to both charm releases kubernetes-master-291 and kubernetes-master-292. If we go back and change that resource, the readme contents won't be updated, and it's likely that it will cause confusion. Different version in the README, vs status output. 

And it's not trivial to go back and update a charm from a specific revision, with that information in the readme. In fact, I don't think it's possible. You would have to publish a new revision of the charm at that revision, with the readme update. thus yielding 3 potential variants, that if not kept under close scrutiny, could cause lots of headache later.

For more details about how this process works, the docs have a good bit of information about the two indpendently moving parts that are interrelated

https://jujucharms.com/docs/2.0/authors-charm-store

**Comment by chuckbutler** [comment]
@rimusz - Are you satisfied with these answers? If not, let's re-open the bug and continue conversation. At this point with 20+ days of no activity I think we're OK to close this as is, as the Kubernetes versions are floating and dependent. We don't plan on shuffling out bins on releases but reserve the right to do so in the event of a botched release.
